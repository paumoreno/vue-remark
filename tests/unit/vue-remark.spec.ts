import VueRemark from "@/vue-remark";
import { mount } from "@vue/test-utils";
import fs from "fs";
import path from "path";
import Shortcode from "../../src/shortcode.vue";

const pathToExample = path.join(__dirname, "..", "..", "src", "example.md");
const exampleMarkdown = fs.readFileSync(pathToExample, {
  encoding: "utf8",
});

describe(VueRemark.name, () => {
  test("snapshot", () => {
    const wrapper = mount(VueRemark, {
      propsData: {
        source: exampleMarkdown,
      },
    });

    expect(wrapper).toMatchSnapshot();
  });

  describe("when providing plugins and renderers", () => {
    // Disabled due to https://github.com/djm/remark-shortcodes/issues/19
    test.skip("correctly applies them", () => {
      const wrapper = mount(VueRemark, {
        propsData: {
          source: [
            "### Using Shortcodes",
            `{{> MailchimpForm id="chfk2" <}}`,
          ].join("\n"),
          plugins: [
            [
              require("remark-shortcodes"),
              {
                startBlock: "{{>",
                endBlock: "<}}",
              },
            ],
          ],
          renderers: {
            shortcode: Shortcode,
          },
        },
      });

      expect(wrapper).toMatchSnapshot();
    });
  });
});
