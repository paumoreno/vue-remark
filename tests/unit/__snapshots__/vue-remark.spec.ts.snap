// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VueComponent snapshot 1`] = `
<div>
  <h1><span>
  Heading 1
</span></h1>
</div>
`;
